### PR TITLE
Add ogg opus training support

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

**Comment by reuben** [comment]
This should allow mixing sample rates by normalizing all training files to `--audio_sample_rate`. Validation and test sets are not normalized by default and should be taken care of manually, I felt like it's better to leave those sets as untouched as possible.

File: training/deepspeech_training/util/audio.py [hunk 1]
```diff
@@ -21,8 +23,9 @@
 AUDIO_TYPE_PCM = 'application/vnd.mozilla.pcm'
 AUDIO_TYPE_WAV = 'audio/wav'
 AUDIO_TYPE_OPUS = 'application/vnd.mozilla.opus'
-SERIALIZABLE_AUDIO_TYPES = [AUDIO_TYPE_WAV, AUDIO_TYPE_OPUS]
-LOADABLE_AUDIO_EXTENSIONS = {'.wav': AUDIO_TYPE_WAV}
+AUDIO_TYPE_OGG_OPUS = 'application/vnd.deepspeech.ogg_opus'
```
Comment (lissyx) [inline training/deepspeech_training/util/audio.py:26]: why do we need this?

File: training/deepspeech_training/util/audio.py [hunk 2]
```diff
@@ -340,6 +346,102 @@ def read_opus(opus_file):
...
+        ctypes.pointer(error)
+    )
+
+    if error.value != 0:
+        raise ValueError(
+            ("Ogg/Opus buffer could not be read."
+             "Error code: {}").format(error.value)
+        )
+
+    channel_count = pyogg.opus.op_channel_count(opusfile, -1)
```
Comment (lissyx) [inline training/deepspeech_training/util/audio.py:365]: so we support multi channel?

File: training/deepspeech_training/util/flags.py [hunk 3]
```diff
@@ -24,6 +24,7 @@ def create_flags():
     f.DEFINE_integer('feature_win_len', 32, 'feature extraction audio window length in milliseconds')
     f.DEFINE_integer('feature_win_step', 20, 'feature extraction window step length in milliseconds')
     f.DEFINE_integer('audio_sample_rate', 16000, 'sample rate value expected by model')
+    f.DEFINE_boolean('normalize_sample_rate', True, 'normalize sample rate of all train_files to --audio_sample_rate')
```
Comment (lissyx) [inline training/deepspeech_training/util/flags.py:27]: Hm, we dont take any risk here?

**Review by lissyx — APPROVED** [review]

File: training/deepspeech_training/util/audio.py (hunk 1 above)
Comment (reuben) [reply training/deepspeech_training/util/audio.py:26]: MLS uses `.opus` files which are specified as `audio/ogg`, which is the same for every Ogg container. But our code only actually handles Ogg with Opus encoded audio inside, which is not `audio/ogg`. In any case it's not like these mime types are an exposed API, I just didn't want to take `audio/ogg` in case one day we move towards supporting proper Ogg containers with more codecs.

File: training/deepspeech_training/util/audio.py (hunk 2 above)
Comment (reuben) [reply training/deepspeech_training/util/audio.py:365]: We've had since the dataset refactor. They get converted to 16kHz mono WAV during the augmentation steps.

File: training/deepspeech_training/util/flags.py (hunk 3 above)
Comment (reuben) [reply training/deepspeech_training/util/flags.py:27]: Risk is that it silently breaks somehow. But due to some TensorFlow limitations it's the only way to handle multiple sample rates in the input files, because the AudioSpectrogram and Mfcc ops take hardcoded parameters (can't be tensors) that depend on the sample rate.

File: training/deepspeech_training/util/audio.py (hunk 2 above)
Comment (lissyx) [reply training/deepspeech_training/util/audio.py:365]: ok, it was unclear in the code here, but indeed I remember seeing the resampling phase

Comment (reuben) [reply training/deepspeech_training/util/audio.py:365]: Yeah it's not super obvious, it happens here: https://github.com/reuben/DeepSpeech/blob/fcbd92d0d75beee36473aa44669fa330ca522cdc/training/deepspeech_training/util/augmentations.py#L165

(That gets called even if no augmentations are used)

Comment (CatalinVoss) [reply training/deepspeech_training/util/audio.py:365]: Hmmm I don't see where `change_audio_type` ends up performing resampling or merges channels. Can you help me out?

Comment (CatalinVoss) [reply training/deepspeech_training/util/audio.py:365]: Hmmm I don't see where `change_audio_type()` ends up performing resampling or merges channels.
